src/moxfm.man
.TH MOXFM 1 "January 1996" "X Version 11"
.SH NAME
moxfm \- Motif file and application manager

.SH SYNOPSIS
.B moxfm
[\fIoptions\fP ...]
.sp
.SH DESCRIPTION
\fIMoxfm\fP is an OSF/Motif based file and application manager for Unix
workstations running the X window system. Moxfm is quite similar to the
Athena widgets based \fIxfm\fP filemanager, and it was developed using
the source code of xfm. Moxfm uses configuration files with the same
format as xfm (except the device configuration file, see CONFIGURATION).

Moxfm \- as xfm \- allows you to browse your directory tree and to copy, move,
link and delete files in an intuitive way by simple drag-and-drop actions. It
provides context-sensitive popup menus. Directory displays are updated
automatically in regular intervals when the contents of the directory
changes. The integrated application manager provides a kind of ``shelf'' on
which you can place your favorite applications, as well as the files and
directories you are currently working with. It also allows you to access
different groups of applications and files. User-definable file types let you
specify a command to be executed when double-clicking on a file or dropping
other files onto it. You can place files and applications as icons on the root
window and use them both for starting applications, opening directory windows,
and as drop sites. Application windows as well as the desktop icons can also
be updated automatically whenever the corresponding application files
change. Furthermore, moxfm provides a monitoring facility which allows to
``have an eye'' on important files and directories and respond with
user-defined actions to certain types of events when the status of one of
these files changes. Last not least, \fImoxfm\fP can automatically mount and
unmount special devices like floppies as you open and close the corresponding
directories (mount points) or copy files to them.

.SH "OPTIONS"
Moxfm accepts all the usual toolkit options. Furthermore, the following
options let you print moxfm's version number and control which windows
are displayed at startup.
.TP 4
.B \-version
Print the version number and exit.
.TP 4
.B \-appmgr
Only display the application manager window and the desktop icons initially.
.TP 4
.B \-filemgr
Only display the file manager window initially. 
.TP 4
.B \-dir \fIpath\fP
Only display a file manager window of \fIpath\fP.
.TP 4
.B \-ignorestartup
Ignore the (optional) startup file, which specifies which windows are opened
at which locations at startup.
.PP
When both \fI-appmgr\fP and \fI-filemgr\fP are specified, the main application
window and a file manager window of the home directory will be displayed,
which is also the default when there is no startup file, or when the
\fI-ignorestartup\fP option is given.
.SH "RUNNING MOXFM FOR THE FIRST TIME"
The first time a user is running moxfm, the program will copy a number of
configuration files into a directory, which is defined by the
\fIapplicationCfgPath\fP resource (usually \fI~/.moxfm\fP), unless this
directory already exists.  These personal configuration files may be edited to
customise moxfm and xfm to the user's taste. In addition, the user might want
to copy the application defaults file \fIMoxfm\fP into his home directory and
customize it.
.SH "USAGE"
Most of it should be fairly obvious. There is an arbitrary number of
application and file windows. Each
.B file window
shows the contents of one directory. In order to perform an action on a
\fIsingle\fP file or directory, you can press the right mouse button over it
to obtain a popup menu, or you press the middle mouse button on it and drag it
to a second file window, a directory icon (e.g.) in the same file window, a
button in an application window, or the background of an application
window. If you want to copy the contents of an ASCII file into e.g. a Motif
editor (such as \fInedit\fP), you can drop the file over the editor window as
well. You can also double-click on an item to start a corresponding action
(like launching an application, editing a file, or changing directories). If
you want to perform an action on \fImultiple\fP files/directories
simultaneously, you can \fIselect\fP multiple files with the left mouse key in
combination with the \fICtrl\fP key. Then you can select an action from the
pulldown file menu, or you can drag the files by pressing the middle mouse
button in one of the selected items and moving the mouse. If you want to
select a number of files in consecutive order, you can select the first file
of the block with the left mouse key and then select the last file with the
left mouse key in connection with the \fIShift\fP key.
.PP
If you drop files and/or directories onto the background of a file window or
onto a directory icon within a file or application window, the
files/directories are \fImoved\fP to the corresponding directory. If you want
to \fIcopy\fP the items instead of moving them, you have to hold the
\fICtrl\fP key pressed when releasing the middle mouse button. You can make
\fIlinks\fP to the files/directories in a different directory by pressing the
\fICtrl Shift\fP keys when releasing the middle mouse key over the
corresponding directory.
.PP
The action taken when double-clicking on a file depends on the type of the
file. If it is a directory, it is displayed in the file window. If it is an
executable, the program is started. Other files are opened in the default
editor (specified by the \fIdefaultEditor\fP resource), unless another action
is given for the type of the file in the \fIxfmrc\fP file (see CONFIGURATION
below).  Moxfm can determine the type of the file by reading the first 1 KB of
its contents and comparing it with the "\fImagic headers\fP" of known file
types (unless moxfm has been compiled without the -DMAGIC_HEADERS option),
and/or by matching the file name with predefined patterns in the \fIxfmrc\fP
file. On slow devices, such as floppies or CD-ROMs, reading the headers of all
files in a directory can take unacceptably long time. Therefore moxfm can be
told to use an alternative \fIxfmrc-nomagic\fP file without magic headers for
those devices (see the description of the \fInoMagicConfigFile\fP resource
below).
.PP
In the \fIxfmrc\fP file you can also specify drop actions for different file
types, so when you drop files onto a file of one of these types the
appropriate action is performed. Applications can be launched by dropping
files on them.  When you drop files onto a file for which there is no drop
action defined, the files are copied, moved or linked to the directory of the
file window. In file windows, valid drop sites are always highlighted when you
drag files over them.
.PP
Directories can be displayed in two different forms: \fIicon view\fP (display
directories and files as icons) and \fItext view\fP (similar to \fIls -l\fP).
These view types are selected from the \fIOptions\fP menu, the default view
type is given by the \fIinitialDisplayType\fP resource.
.PP
There are several ways to change the directory displayed in a file window. You
can directly edit the text field which displays the name of the shown
directory and press the \fIEnter\fP key, or you can double-click on a
directory icon.  To go to your home directory you can also select the \fIGo
home\fP item from the \fIDirectory\fP menu or simply press the button with the
symbol of a (typical swedish) house; to go to the parent directory you can
select the \fIGo up\fP item from the \fIDirectory\fP menu or press the button
with the parent directory symbol. You can change into one of the current
directory's ancestor directories by pressing the right mouse button over the
directory text field and choosing a directory from the posted popup menu.
.PP
The other buttons in the file windows have the following meanings. When you
press the button with the glasses symbol, the directory contents will be
reread and redisplayed. Clicking on the button with two directory symbols on
it opens a second window of the same directory. Pressing the button which
shows the symbol of a window with colored icons will cause a new instance of
the main application window to come up. When you click on the button with the
green leftarrow, all your changes to the text field containing the directory
name are discarded (unless you have already pressed the \fIEnter\fP key).
.PP
The
.B application windows
contain arbitrary numbers of buttons, for which a push and/or a drop action
can be defined. You can change the actions and the icon of a button by
pressing the right mouse button over it and selecting \fIEdit\fP from the
popup menu. For the syntax of the push action and drop action fields, see FILE
TYPE CONFIGURATION below. To create a new button, press the right mouse button
over the background of the application window and select the \fIInstall
application...\fP item, or drag a file or directory from a file window to the
background of the application window. In this case a new button is created,
taking the push and drop actions for the type of the file as the push and drop
actions for the button. Clicking on a directory button in an application
window will start a file window showing the contents of the directory. You can
also copy and move buttons between different application windows or within an
application window by drag'n'drop with the middle mouse button. If the
\fICtrl\fP key is pressed when releasing the middle mouse button, the button
is copied, otherwise it is moved.
.PP
You can also drop other types of objects, such as text strings, on the
application window. Users may define the object types which can be dropped on
application windows in the \fItargets\fP configuration file (see below). For
each object dropped onto an application window a button is created, which can
be used as a drag source.  The push action of the button is specified in the
\fItargets\fP file.
.PP
There is one \fImain application group\fP, which is specified by the
\fImainApplicationFile\fP resource. An application window with this main menu
is opened e.g. when you start moxfm using the \fI-appmgr\fP option. Each
application window may contain buttons for further application groups. When
you press such a button, the application window will show the buttons of that
group instead.  When you select the \fIOpen\fP item from the popup menu of
such a button, a new application window showing the selected application group
is opened.
.PP
You can place files and applications as icons on the root window
by choosing the \fIPut on desktop\fP item from their popup menus. For these
\fIdesktop icons\fP a push action and a drop action can be defined, as for
application buttons. To invoke the push action of a desktop icon you have to
double-click on it. To change the actions you can select \fIEdit\fP from the
icon's popup menu.

.SH "MENU COMMANDS"
.sp
FILE MENU
.PP
File manipulation operations.
.TP 4
.I New...
.br
Create a new (and empty) file.
.TP 4
.I Copy...
.br
Create a copy of a single item with a new name or copy selected items to
another directory.
.TP 4
.I Move...
.br
Rename a single item (directory or file) or move selected items to another
directory.
.TP 4
.I Link...
.br
Like \fICopy\fP, but creates symbolic links rather than copying the selected
items.
.TP 4
.I Perform action...
.br
You are prompted for an action to be performed on the selected items.
.TP 4
.I Delete
.br
Delete the selected items.
.TP 4
.I Select...
.br
Select items by pattern. The usual metacharacters are recognized (\fI*\fP,
\fI?\fP, \fI[ ]\fP). (Currently there is no provision for escaping these.)
.TP 4
.I Select all
.br
Select all items in the current directory (except the parent directory).
.TP 4
.I Deselect all
.br
Deselect all items.
.TP 4
.I Edit config file
.br
The \fIxfmrc\fP configuration file is loaded into the default editor.
.TP 4
.I Edit monitor file
.br
The \fIxfmmon\fP file (see CONFIGURATION) is loaded into the default editor.
.TP 4
.I Reload configuration
.br
The \fIxfmrc\fP, xfmmon and \fImagic\fP configuration files are reread and
all file and application windows are updated.
.TP 4
.I About moxfm...
.br
A message box with some information about moxfm (copying etc.)
is popped up.
.TP 4
.I Close window
.br
Close the current file window
.TP 4
.I Exit.
.br
Terminate moxfm.
.PP
DIRECTORY MENU
.PP
Operations dealing with directories and the file window.
.TP 4
.I Make directory...
.br
Create a new directory.
.TP 4
.I Filesystem info...
.br
Show information on free space and filesystem parameters.
.TP 4
.I Reread files.
.br
Reread and redisplay directory contents.
.TP 4
.I Filter files...
.br
Only show files whose names match the given pattern. The usual metacharacters
are recognized (\fI*\fP, \fI?\fP, \fI[ ]\fP). (Currently there is no provision
for escaping these.) This only affects normal files, i.e. directory items will
not be filtered. The \fIClear\fP button in the Filter dialog form reverts to
the full display.
.TP 4
.I Go home
.br
Change to your home directory.
.TP 4
.I Go up
.br
Go to the parent directory.
.TP 4
.I Enter command...
.br
You are prompted for a command, which will be interpreted by your default
shell in the directory of the file window.
.PP
WINDOWS MENU
.PP
Operations concerning windows.
.TP 4
.I New file window
.br
Open another window on the current directory.
.TP 4
.I New application window
.br
Open a window with the main application menu.
.TP 4
.I Mount table window
.br
Open a window which displays the configured mountable devices (if not yet
present).
.TP 4
.I Raise desktop icons
.br
Raise the desktop icons to the top.
.TP 4
.I Start xterm
.br
Start an xterm from the current directory.
.TP 4
.I Save windows
.br
Save positions and states of all moxfm's windows to the startup file. The
desktop icons are saved, too.
.TP 4
.I Save desktop icons
.br
The desktop icons and their positions are saved.
.PP
OPTIONS MENU
.PP
Options for the directory display.
.TP 4
.I Icon view
.br
Select the icons form display.
.TP 4
.I Text view
.br
Select the text form display.
.TP 4
.I Sort by name
.br
Sort directory by name.
.TP 4
.I Sort by size
.br
Sort directory by size.
.TP 4
.I Sort by date.
.br
Sort directory by date.
.TP 4
.I Show hidden files.
.br
Show hidden files (files starting with a dot).
.TP 4
.I Show directories
.br
Show directory items.
.TP 4
.I Directories first
.br
Show directories first in the window.
.TP 4
.I Use magic headers
Determine the file types by reading the first 1 KB of each file and comparing
them with the \fImagic headers\fP of known file types. Switch this off, if
file type recognition is too slow. This requires that you have specified an
alternative \fIxfmrc-nomagic\fP file (see \fInoMagicCfgFile\fP resource).
.PP
FILE POPUP MENU
.PP
Operations on a single file. This menu pops up when pressing the right mouse
button on a directory or file icon. Not all of these operations are available
for all filetypes.
.TP 4
.I Open
.br
Open a file window on the selected item. This option is only available if the
selected item is a directory.
.TP 4
.I Execute
.br
Launch the selected application. You are prompted for command line
parameters. This option is only available for executables.
.TP 4
.I Edit
.br
Edit the selected item using the program specified in the \fIdefaultEditor\fP
resource (only available if the selected item is not a directory).
.TP 4
.I View
.br
Same as Edit, but invokes a program for viewing the file (\fIdefaultViewer\fP
resource).
.TP 4
.I Perform action...
.br
You are prompted for an action to perform on the selected item.
.TP 4
.I Rename...
.br
Rename the selected item.
.TP 4
.I Copy...
.br
Copy the selected item.
.TP 4
.I Move...
.br
Same as rename, but the old name is not displayed in the dialog.
.TP 4
.I Link...
.br
Create a symbolic link to the selected item.
.TP 4
.I Delete
.br
Delete the selected item.
.TP 4
.I Properties...
.br
Display information about the selected item (file size, permissions and
such). You can change permissions and ownership.
.TP 4
.I Put on desktop
.br
Place the selected item as an icon on the root window.
.PP
APPLICATION GROUP MENU
.PP
Operations for managing the application window. This menu pops up when
pressing the right mouse button on the background of an application window.
.TP 4
.I Install application...
.br
Install a new application in the application window. Pops up a dialog form
into which you can enter the necessary information (see APPLICATION FILES for
a discussion of the fields in this form).
.TP 4
.I New application group...
.br
Simplified install dialog form for creating a new application box and making it
accessible from the current application menu (see APPLICATION FILES).
.TP 4
.I Save setup
.br
Save the configuration of the current application window.
.TP 4
.I Reload setup
.br
Reload the configuration of the current application window from the
configuration file and discard all recent changes.
.TP 4
.I Edit setup file
.br
Open the configuration file of the current application window with the default
editor.
.TP 4
.I Windows
.br
A cascade menu with items listed below is posted.
.TP 4
.I Close window
.br
Close the current application window.
.TP 4
.I Exit
.br
Terminate moxfm.
.PP
Windows Cascade Menu
.TP 4
.I Main appl window
.br
Open a new application window with the main apllication menu.
.TP 4
.I Open file window
.br
Open a file window displaying your home directory.
.TP 4
.I Mount table window
.br
Open a window which displays the configured mountable devices (if not yet
present).
.TP 4
.I Raise desktop icons
.br
Raise the desktop icons to the top.
.TP 4
.I Save windows
.br
Save positions and states of all moxfm's windows to the startup file. The
desktop icons are saved, too.
.TP 4
.I Save desktop icons
.br
The desktop icons and their positions are saved.
.PP
APPLICATION POPUP MENU
.PP
Operations on a single application item (button). This menu pops up when
pressing the right mouse button on a button in the application window.
.TP 4
.I Open
If the selected item represents an application menu, a new application window
displaying this menu is opened. Otherwise, the default push action of the
selected button is invoked.
.TP 4
.I Edit...
.br
Edit an application item. Pops up a dialog form which allows you to change
the configuration information associated with the selected item (see
CONFIGURATION for a discussion of the fields in this form).
.TP 4
.I Delete
.br
Remove the selected button from the application window.
.TP 4
.I Put on desktop
.br
Place the selected item as an icon on the root window.
.PP
DESKTOP ICON POPUP MENU
.PP
Operations on a single desktop icon. This menu pops up when pressing the right
mouse button on a desktop icon.
.TP 4
.I Open
If the selected icon represents an application menu, a new application window
displaying this menu is opened. Otherwise, the default push action of the
selected icon is invoked.
.TP 4
.I Edit...
.br
Edit a desktop icon. Pops up a dialog form which allows you to change
the configuration information associated with the selected icon (see
CONFIGURATION for a discussion of the fields in this form).
.TP 4
.I Save icon
.br
Save the position and state of the selected icon (only available if the
autoSave resource is false, see RESOURCES).
.TP 4
.I Remove from desktop
.br
Remove the selected icon from the root window.
.PP
.SH "RESOURCES"
Various aspects of moxfm can be configured by changing corresponding
resource settings in the application defaults file. Some important resources
are listed below:
.TP 4
.B bitmapPath
.TP 4
.B pixmapPath
.br
The path on which to search for bitmap and pixmap icons, respectively.
Different directories are separated by a colon.
.TP 4
.B mainApplicationFile
.TP 4
.B configFile
.TP 4
.B noMagicConfigFile
.TP 4
.B startupFile
.TP 4
.B DTIconFile
.TP 4
.B devFile
.TP 4
.B userObjectFile
.TP 4
.B magicFile
.br
The names of the application and configuration files used by \fImoxfm\fP (see
CONFIGURATION). Normally, these files will be located in \fI~/.moxfm\fP.  You
may wish to change this, e.g., if you want to provide a system-wide
\fIxfmrc\fP file. (The application files should always be kept in the user's
home directory, such that each user can save his/her private application
settings.) You may specify these pathnames as \fI~/path\fP, but not as
\fI~user/path\fP.
.TP 4
.B applicationCfgPath
.br
The directory in which the application files for new application groups are
located (see the \fIInstall group\fP option of the application menu), usually
\fI~/.moxfm\fP.
.TP 4
.B tmpObjectDir
.br
This resource points to the directory where temporary object files are created.
These files contain the data of user defined objects dropped on application
windows and are deleted when the corresponding buttons are removed.
.TP 4
.B updateInterval
.br
Set the time interval in milliseconds in which to perform automatic folder
(and application window) updates. Default: 10000.
.TP 4
.B doubleClickTime
.br
Set the time interval in milliseconds for which a sequence of two mouse clicks
should be interpreted as a double click. Default: 300.
.TP 4
.B firstIconPos
.br
Specifies the position of the first desktop icon. Notation:
[+-]\fIx\fP[+-]\fIy\fP.  The default is \fI+0+0\fP (upper left corner of
screen).
.TP 4
.B iconAlignment
.br
This resource determines the direction of the desktop icons' alignment
relative to the first desktop icon. Allowed values are \fIUp\fP, \fIDown\fP,
\fILeft\fP and \fIRight\fP. \fIDown\fP (which is the default) e.g. means that
the second desktop icon is placed \fIunder\fP the first, the third under the
second, etc.
.TP 4
.B iconsOverrideRedirect
.br
Some mwm-incompatible window managers such as ol[v]wm and twm
won't display desktop icons correctly (e.g., the window borders may
not be completely removed). In this case you might want to set this
resource to true, which makes the window manager ignore desktop
icons. This might cause strange side effects, esp. when you are moving
icons.
.TP 4
.B checkFiles
.br
Set this to true (default) if you want the time stamps of all files (except
subdirectories) shown by the file windows to be checked on automatic folder
updates. If set to false, only the timestamp of the directory itself is
checked.
.TP 4
.B checkApplicationFiles
.br
Set this to true (default) if you want application windows to be updated
automatically whenever the time stamp on the application file changes. (This
option is ignored when the autoSave resource is set to false, see below.)
.TP 4
.B autoSave
.br
When you change the contents of an application group, or create, remove or
change a desktop icon, moxfm will automatically save the changes to the
configuration file, if this resource is set to true. Otherwise, you are asked
if you want to save your changes when you close the application window or
change the application group. Set this resource to true if you want to enable
automatic updates of application files (see above).
.TP 4
.B saveOnExit
.br

ChangeLog
2001-09-03  root  <Dr.Graef@t-online.de>

	* changed program info (version number, www pointer)

	* fixed a bug in FmFw.c (eliminate trailing newline from ctime
	return value)

	* moxfm config files now go to .moxfm

	* face-lifting using modern apps and 3dicons, also tweaked some of
	the resources

	* added magic.c patches (from xplore/xfm)

	* added Bob Woodside's 1.0.1 patches (see below)

2001-08-15:	Version 1.0.1:

		New features:
                None.

                Changes:
                This is a minimal maintenance release designed to correct a
                few problems that have cropped up since Oliver left off work
                on moxfm, and produce a viable current distribution of the
                source.

                Bug Fixes:
                - regexp/regexp.c: Commented out argument-less declaration for
                strchr(), which causes compile failures with recent compilers.
                I know of no systems on which this declaration is needed any
                longer.
                - regexp/regsub.c: Commented out argument-less declaration for
                strncpy().
                - src/Common.h: Added logic to force define for SYSV on recent
                Linux systems, so that getwd will be properly redefined as
                getcwd. getwd is a dangerous function that generally should not
                be used, and recent versions of gcc issue a warning to this
                effect.
                - src/FmFw.c, src/FmIcons.c: Added a workaround for a failure
                of double-clicks to work under certain window managers/window
                manager modes, due to ill-synchronized passive grabs by the
                window manager on a parent window. This "fix" is based on a
                similar one made by Till Straumann for xfm.

**************
OLD ChangeLog:
**************

Date (yy-mm-dd):

95-08-17:	Moxfm v. 0.9 is available for public
		use for the first time via WWW. It is
		based on xfm v. 1.3.2.

95-08-21:	Version 0.9pl1: Bugfixes concerning 
		the state of the mount table window at
		startup 

95-08-22:	Version 0.91: Changes in function
		fnmultimatch() to allow for escaping
		blanks.

95-08-24:	Version 0.92: Changes in file window
		update code.

95-08-25:	Version 0.93: Code for updating file
		windows changed again. Mount point
		detection improved.

95-08-26:	Version 0.94: When copying, moving,
		linking or deleting fails for a number
		of files, the screen is no longer flooded
		with error message dialogs, but the
		user has the chance to cancel the
		operation. Changes in the code for
		recursive copying. Bugfixes.

95-08-28:	Version 0.94pl1: Bugfix concerning
		status display in connection with file
		selections.

95-08-29:	Version 0.94pl2: Bugfix in filesystem
		info dialog.

95-08-30:	Version 0.95: Changes in FmDragDrop.c
		in order to make moxfm compatible
		with future versions which (hopefully)
		will support copying files to/from a
		remote machine. Bugfix concerning
		-dir command line argument.

95-08-31:	Version 0.95pl1: Fixes in device mounting
		in FmFwActions.c and FmOps.c.

95-09-02:	Version 0.96: Major code clean-up of
		FmFwActions.c, also affecting other
		modules. Bugfixes here and in FmOps.c.

95-09-08:	Version 0.97: FIRST VERSION UPLOADED TO
		FTP-SERVER.
		
		- You cannot drop a file onto itself
		  any longer.
		
		Bugfixes:
		- When a filename is specified for an
		  application button, the filename is
		  passed as first argument also for
		  drop actions. This implies e.g. that
		  the drop action for the trashcan should
		  be
		  "shift;mv -f -b -V numbered $* ~/.trash"
		  now (as for xfm).
		- The icon names are set correctly now.

95-09-09:	Bugfix in usermount utility: userfstab
		file is checked for all mount options,
		not only "read-only" now.

95-09-18:	Version 0.98:

		New features:
		- There is a popup menu of the directory
		  name text field now (thanks to Steve
		  Blake for idea.)
		- The "File" pulldown menu has a "Perform
		  action" item now.
		- The mount table window has a "Close"
		  button.
		- Links are always made to *relative*
		  paths.
		- New resources saveOnExit and newWinOnDirPopup.
		
		Bugfixes:
		- Moxfm should hopefully compile on OSF/1
		  and SVR4 systems now (thanks to Albert
		  Graef and Volker Mertens).
		- Moxfm compiles with Motif 2.0 now.
		- Moxfm doesn't crash any longer when it
		  cannot allocate enough colors for the
		  pixmaps (thanks to Detlef Schmier).
		- The icon names should really be correct
		  now.
		  
		Changes:
		- When the autoSave feature of the
		  application windows is enabled, the
		  configuration is saved directly after
		  every change and not first when moxfm
		  exits.
		- The *relative* paths of dropped files
		  are passed to application buttons with
		  specified directories now.
		- There is a "Perform action" item in the
		  pulldown menu of directory icons, too.
		- New suppressWarnings resource instead of
		  suppressXtWarnings resource.
		- The names of the application group
		  configuration files don't start with "xfm-"
		  any longer.

96-05-16:	Version 1.0:

		New features:
		- Files and applications can be placed as icons
		on the root window.
		- Added automatic application window updates along
		with a checkApplicationFiles resource, and an applBoxIcon
		resource for specifying the standard application group
		icon. Furthermore, newly created application groups now
		point back to the group they were created in.
		(ag@muwiinfa.geschichte.uni-mainz.de)
		- Added a facility for monitoring files during
		automatic updates, along with the new xfmmon
		configuration file (AG).
		- You can drop user defined objects (listed in
		targets configuration file) on application windows.
		- You can switch off magic headers for certain
		devices and file windows.
		- You can move files across devices now.
		- The file type configuration file can be reread
		during runtime.

lib/Moxfm.cpp
!***********************************************************************
! Application defaults for moxfm.
! Most of these are the defaults anyway, but they are listed here so you can
! change them if necessary.
!***********************************************************************

! Search paths for icon files
*bitmapPath: BITMAPDIR
*pixmapPath: PIXMAPDIR

! Default applications file to be loaded at startup
*mainApplicationFile:	~/.moxfm/Main

! Configuration file (where the file types are specified)
*configFile: ~/.moxfm/XFMRC
#ifdef MAGIC_HEADERS
! If you want to have the possibility *not* to use magic headers
! for *certain* files, you must here specify a file type configuration
! file which does not use magic headers
*noMagicConfigFile:	~/.moxfm/xfmrc.nomagic
#endif
! Startup file (determines, which windows are opened at startup
! if none of the options -ignorestart, -appmgr, -filemgr or -dir
! are specified)
*startupFile:		~/.moxfm/startup

! Device configuration file (floppies and such)
*devFile:		~/.moxfm/moxfmdev

! The monitor config file (specifies files to be monitored during automatic
! updates)
*monitorFile:		~/.moxfm/xfmmon

! Directory where application files are to be created
*applCfgPath:		~/.moxfm/

! File which contains user defined object types
*userObjectFile:	~/.moxfm/targets

! Directory where temporary objects are to be created
*tmpObjectDir:		~/.moxfm/tmpobj

! Magic headers configuration file (files you recognize by their header)
*magicFile:		~/.moxfm/magic

! Standard icon for new application groups
*applBoxIcon:		stuff.xpm

! (Don't) save application files and desktop icons
! automatically on changes:
*autoSave: 		true

! (Don't) save window positions and contents automatically
! on exit:
*saveOnExit: 		false

! Double click time in milliseconds
*doubleClickTime:		300

! Time interval in milliseconds for automatic folder and application window
! updates (set to zero to disable this feature)
*updateInterval:		10000

! Set this to True, if every single file (not only the directory) shall be
! checked for changes on automatic folder updates
*checkFiles:			true

! Set this to True if application windows should be updated automatically
! Works only if Moxfm*autoSave is set to true!
*checkApplicationFiles:		true

!***********************************************************************
! Preferences
!***********************************************************************

! Geometry for application and initial file manager windows
*initAppGeometry:		332x358-95+230
*initFileGeometry:		509x448+69+136
! width x height for subsequent file windows
*fileGeometry:			500x400
! width x height for subsequent application windows
*appGeometry:			300x350

! Position of the first desktop icon:
*firstIconPos:			+0+0
! Alignment of desktop icons (row/column with start
! at the position of the first icon). Values: Up, Down, Left, Right
*iconAlignment:			Down

! NOTE: The background color of icons is determined from the Moxfm.background
! resource. Make sure that this resource is always given a value appropriate
! for *both* application and file windows.

! The background color for file and application windows
*background:			#8F5BA8F5C7AD
*menubar*background:		#91EABFFFD998

! The background color for desktop icon labels:
*dticonColor:			#8F5BA8F5C7AD

*XmTextField*background:	white
!*menubar*foreground:		black

*menubar*tearOffModel:	TEAR_OFF_DISABLED

! This color is used when an icon button
! is highlighted or marked
*selectColor:		red

! Specify colors for display of filenames:
*filenameColor:		black
*linknameColor:		maroon
! This color applies only to text display style
*textDirColor:		lemon chiffon
! Highlight color for desktop icons
*dropColor:		lemon chiffon

*file window*form.highlightColor:	yellow
*file window*icon box*highlightColor:	yellow
#ifdef SGI_MODE
*sgiMode: true
*XmScrolledWindow.scrolledWindowMarginWidth:	3
*XmScrolledWindow.scrolledWindowMarginHeight:	3
#endif
! Use these only if you  have the 75dpi fonts. (recommended)
*boldFont:			-adobe-helvetica-bold-r-*-*-12-*
*iconFont:			-adobe-helvetica-medium-r-*-*-10-*
*buttonFont:			-adobe-helvetica-medium-r-*-*-12-*
*menuFont:			-adobe-helvetica-medium-r-*-*-12-*
*labelFont:			-adobe-helvetica-medium-r-*-*-12-*
*statusFont:			-adobe-helvetica-medium-r-*-*-12-*
*ttyFont:			-adobe-courier-medium-r-*-*-12-*

! This is for file names in text display mode:
*filenameFont:			-adobe-courier-medium-r-*-*-12-*

! This must be a fixed width font
*cellFont:			fixed

*XmMessageBox*fontList:		-adobe-helvetica-medium-r-*-*-12-*
*XmSelectionBox*fontList:	-adobe-helvetica-medium-r-*-*-12-*
*XmPushButtonGadget*fontList:	-adobe-helvetica-medium-r-*-*-12-*
*Popup*fontList:		-adobe-helvetica-medium-r-*-*-12-*
*menubar*fontList:		-adobe-helvetica-bold-r-*-*-14-*
*menubar*Pulldown*fontList:	-adobe-helvetica-bold-r-*-*-12-*

! Specify the sizes of the different icon toggles. The following values
! are appropriate for the icons supplied with xfm.
*appIconWidth:		48
*appIconHeight:		48
*fileIconWidth: 	48
*fileIconHeight:	40

*Editform*XmTextField*width:	 300

! Confirmation for various operations
*confirmDeletes: 		true
*confirmDeleteFolder: 		true
*confirmCopies: 		false
*confirmMoves: 			true
*confirmLinks:			false
*confirmOverwrite:		true
*confirmQuit:			true

! If you are disturbed by frequent Xt warnings (type "actions
! not found"), you might want to set this to true:
*suppressWarnings:		false

! Set this to true so you can see when moxfm is copying files
*showCopyInfo:			true

! Echo actions on stderr (useful for debugging purposes)
*echoActions:			false

! Directory display in Text type
*showOwner: 			true
*showDate: 			true
*showPermissions: 		true
*showLength: 			true

! Open a new window when a new directory is selected from
! the parent directory popup menu:
*newWinOnDirPopup:		true

! Show hidden files:
*showHidden:			false
! Show directories first (default):
*dirsFirst:			true

! The type of the first and subsequent file windows
! valid values are Icons and Text
*initialDisplayType:	 	Icons

! The type of sorting used by default
! valid values are SortBy{Name,Size,Date}
*defaultSortType: 		SortByName

! This defines the naming policy for files dropped on
! an application button. There are three valid values:
! FileName (the drop action is performed in the directory
! of the dropped files), RelativePath (path relative to the
! directory specified for the application button; default) and
! AbsolutePath.
!*passedNamePolicy:		RelativePath

! The default editor to use
*defaultEditor: 		exec emacs

! The default viewer to use
*defaultViewer:			exec xless

! The xterm to call
*xterm:				exec xterm -sb -sl 500 -j

*startFromXterm:		false
*keepXterm:			false

!***********************************************************************
! Miscellaneous settings (see also FmMain.c)
! Normally you won't have to change these.
!***********************************************************************

*borderWidth: 0
*DragReceiverProtocolStyle: DRAG_PREFER_PREREGISTER
